google/generativeai/answer.py

```diff
@@ -0,0 +1,225 @@
+            single question to answer. For multi-turn queries, this is a repeated field that contains
+            conversation history and the last `Content` in the list containing the question.
+        grounding_source: Sources in which to grounding the answer.
+        answer_style: Style for grounded answers.
+        safety_settings: Safety settings for generated output. Defaults to None.
+        temperature: The temperature for randomness in the output. Defaults to None.
+
+    Returns:
+        Call for glm.GenerateAnswerRequest().
+    """
+    model = model_types.make_model_name(model)
+
+    contents = content_types.to_contents(contents)
+
+    if safety_settings:
+        safety_settings = safety_types.normalize_safety_settings(
+            safety_settings, harm_category_set="new"
+        )
+
+    grounding_source = _make_grounding_passages(grounding_source)
+
+    if answer_style:
+        answer_style = to_answer_style(answer_style)
+
+    return glm.GenerateAnswerRequest(
+        model=model,
+        contents=contents,
+        inline_passages=grounding_source,
+        safety_settings=safety_settings,
+        temperature=temperature,
+        answer_style=answer_style,
+    )
+
+
+def generate_answer(
+    *,
+    model: model_types.AnyModelNameOptions = DEFAULT_ANSWER_MODEL,
+    contents: content_types.ContentsType,
+    inline_passages: GroundingPassagesOptions,
+    answer_style: AnswerStyle | None = None,
+    safety_settings: safety_types.SafetySettingOptions | None = None,
+    temperature: float | None = None,
+    client: glm.GenerativeServiceClient | None = None,
+) -> answer_types.Answer:
+    """
+    Calls the API and returns a `types.Answer` containing the answer.
+
+    Args:
+        model: Which model to call, as a string or a `types.Model`.
+        question: The question to be answered by the model, grounded in the
+                provided source.
+        grounding_source: Source indicating the passages in which the answer should be grounded.
+        answer_style: Style in which the grounded answer should be returned.
+        safety_settings: Safety settings for generated output. Defaults to None.
+        client: If you're not relying on a default client, you pass a `glm.TextServiceClient` instead.
+
+    Returns:
+        A `types.Answer` containing the model's text answer response.
+    """
+    request = _make_generate_answer_request(
+        model=model,
+        contents=contents,
+        grounding_source=inline_passages,
+        safety_settings=safety_settings,
+        temperature=temperature,
+        answer_style=answer_style,
+    )
+
+    return _generate_answer_response(client=client, request=request)
+
+
+@string_utils.prettyprint
+@dataclasses.dataclass(init=False)
+class Answer(answer_types.Answer):
+    def __init__(self, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+        self.result = None
+        if self.answer:
```

Comment:
This should have the same quick access properties as `GenerateContentResponse` > `.parts` and `.text`.